eth/db/chain.py

```diff
@@ -245,6 +252,7 @@ def _set_as_canonical_chain_head(self, header: BlockHeader) -> Tuple[BlockHeader
                 break
             else:
                 old_header = self.get_block_header_by_hash(old_hash)
+                old_headers.append(old_header)
                 for transaction_hash in self.get_block_transaction_hashes(old_header):
                     self._remove_transaction_from_canonical_chain(transaction_hash)
                     # TODO re-add txn to internal pending pool (only if local sender)
```

Comment:
When you update this branch would you mind removing this `TODO` comment as well as the following `pass` statement which is no longer needed.

Reply:
I guess the TxPool will have to track this information and handle it internally. Maybe an event should get broadcast to plugins with the new and orphaned header hashes.

Reply:
:+1: to broadcasting across the plugin event bus.

Reply:
not quite sure how to do this but would be willing to try given a bit more information